Installation issue: xfsprogs

### Steps to reproduce the issue

```console
$ spack spec -I <spec>
Input spec
--------------------------------
 -   xfsprogs

Concretized
--------------------------------
 -   xfsprogs@5.11.0%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]      ^gcc-runtime@13.1.0%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]      ^gettext@0.22.5%gcc@13.1.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]          ^bzip2@1.0.8%gcc@13.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-aarch64
[+]              ^diffutils@3.10%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]          ^libiconv@1.17%gcc@13.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu22.04-aarch64
[+]          ^libxml2@2.10.3%gcc@13.1.0+pic~python+shared build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]          ^ncurses@6.5%gcc@13.1.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu22.04-aarch64
[e]          ^tar@1.34%gcc@13.1.0 build_system=autotools zip=pigz arch=linux-ubuntu22.04-aarch64
[+]          ^xz@5.4.6%gcc@13.1.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu22.04-aarch64
[e]      ^glibc@2.35%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
[e]      ^gmake@4.3%gcc@13.1.0~guile build_system=generic patches=599f134 arch=linux-ubuntu22.04-aarch64
[+]      ^gnuconfig@2022-09-17%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]      ^libinih@master%gcc@13.1.0~strip build_system=meson buildtype=release default_library=shared arch=linux-ubuntu22.04-aarch64
[+]          ^meson@1.3.2%gcc@13.1.0 build_system=python_pip patches=0f0b1bd arch=linux-ubuntu22.04-aarch64
[+]              ^py-pip@23.1.2%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]              ^py-setuptools@69.2.0%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]              ^py-wheel@0.41.2%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]              ^python-venv@1.0%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]          ^ninja@1.12.0%gcc@13.1.0+re2c build_system=generic arch=linux-ubuntu22.04-aarch64
[+]              ^re2c@3.0%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]      ^util-linux@2.40.2%gcc@13.1.0~bash build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]          ^libxcrypt@4.4.35%gcc@13.1.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu22.04-aarch64
[e]              ^perl@5.34.0%gcc@13.1.0~cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu22.04-aarch64
[e]          ^pkg-config@0.29.2%gcc@13.1.0+internal_glib build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]          ^python@3.11.9%gcc@13.1.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu22.04-aarch64
[+]              ^expat@2.6.2%gcc@13.1.0+libbsd build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]                  ^libbsd@0.12.2%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]                      ^libmd@1.0.4%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]              ^gdbm@1.23%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]              ^libffi@3.4.6%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]              ^openssl@3.3.1%gcc@13.1.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu22.04-aarch64
[+]                  ^ca-certificates-mozilla@2023-05-30%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-aarch64
[+]              ^readline@8.2%gcc@13.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-aarch64
[+]              ^sqlite@3.43.2%gcc@13.1.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]          ^zlib-ng@2.2.1%gcc@13.1.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu22.04-aarch64
[+]      ^util-linux-uuid@2.40.2%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-aarch64
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Error: xfsprogs-5.11.0-xo5pncucy7rhbewnu365rfapk3mxkiko: ProcessError: Command exited with status 2:
    '/usr/bin/make' '-j6' 'V=1'
==> Error: xfsprogs-5.11.0-xo5pncucy7rhbewnu365rfapk3mxkiko: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
</pre></details>


### Information on your system

* **Spack:** 0.23.0.dev0 (0ae66e0724fff79f29abb1e780203ad32a005db2)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-aarch64

### Additional information

The package complains about an undefined reference to a function from libintl. This library is part of gettext, and I can confirm that it's present in my environment and has the symbol.

Note that I tried to add a newer version of xfsprogs (version 6.11.0) but the installation fails even earlier (I think the configure script just exits with a non-zero return code).

[spack-build-out.txt](https://github.com/user-attachments/files/17612259/spack-build-out.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/17612260/spack-build-env.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/xfsprogs/package.py
+++ b/var/spack/repos/builtin/packages/xfsprogs/package.py
@@ -24,7 +24,7 @@ class Xfsprogs(AutotoolsPackage):
     depends_on("c", type="build")  # generated
 
     depends_on("libinih")
-    depends_on("gettext")
+    depends_on("gettext@:0.21.1")
     depends_on("uuid")
     depends_on("util-linux")
 
@@ -32,7 +32,7 @@ class Xfsprogs(AutotoolsPackage):
         if name == "cflags":
             if self.spec.satisfies("@:5.4.0 %gcc@10:"):
                 flags.append("-fcommon")
-        elif name == "ldlibs":
+        elif name == "ldlibs" or name == "ldflags":
             if "intl" in self.spec["gettext"].libs.names:
                 flags.append("-lintl")
         return build_system_flags(name, flags)

--- a/var/spack/repos/builtin/packages/xfsprogs/package.py
+++ b/var/spack/repos/builtin/packages/xfsprogs/package.py
@@ -24,7 +24,8 @@ class Xfsprogs(AutotoolsPackage):
     depends_on("c", type="build")  # generated
 
     depends_on("libinih")
-    depends_on("gettext@:0.21.1")
+    depends_on("gettext")
+    depends_on("gettext@:0.21.1", when="@:6.3")
     depends_on("uuid")
     depends_on("util-linux")
 

